### DISCUSSION TITLE
How to always keep a few blank lines under the prompt?

### DISCUSSION THREAD

**Discussion by vvkdev** [fish-shell/fish-shell#10094] [Q&A] [answered]
In bash I used PS1 for this, but what about fish?
Thank you

**Comment by faho** [comment]
https://fishshell.com/docs/current/faq.html#how-do-i-set-my-prompt

In this specific case, you can just add `echo` at the end:

```fish
function fish_prompt
    # Your prompt here, this is just a quick example:
    echo "$PWD>"
    # Now add empty lines:
    echo
    echo
    echo
    # or, to add an arbitrary number:
    string repeat -n 5 \n
end
```

**Reply by vvkdev** [reply]
I meant the lines after the cursor and not between the prompt and the cursor. So that the prompt is never at the very bottom of the window

**Reply by faho** [reply, marked as answer]
You can *try* a postexec handler that prints X lines and then moves the cursor up:

```fish
function make_blank_lines --on-event fish_postexec
    string repeat -n 5 \n
    tput cuu 5
end
```

Put that in config.fish. From a cursory glance it seems to work.

**Comment by vvkdev** [comment]
@faho, why was the discussion closed without an answer to the question? Maybe someone can help me...
